libraries/core_libs/network/include/network/tarcap/packets_handler.hpp

```diff
@@ -35,4 +43,9 @@ class PacketsHandler {
   std::unordered_map<SubprotocolPacketType, std::shared_ptr<PacketHandler>> packets_handlers_;
 };
 
+template <typename PacketHandlerType>
+std::shared_ptr<PacketHandlerType> PacketsHandler::getSpecificHandler(SubprotocolPacketType packet_type) {
+  return std::static_pointer_cast<PacketHandlerType>(getSpecificHandler(packet_type));
+}
```

Comment:
would it be possible to have some kind of table so we do not need to specify both? packet type and packet type handler? or maybe `PacketHandlerType` could have some kind of static variable or not sure how it is handled and based on that we would get the type? `PacketHandlerType::packet_type` ???

Reply:
yes I was thinking about it too, there could be a big switch and thats about it - not sure what is better solution, have this big switch or just specify the type ?

Reply:
@MatusKysel @kstdl fixed, please re - review :)